extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -76,14 +94,26 @@ export class ScrollManager {
   addScrollHandler(handler) {
     // Trigger an event to fire events that might have already happened.
     const size = this.viewport_.getSize();
+    const layoutRect = this.viewport_.getLayoutRect(this.root_);
+    // Cache the scroll height/width for use with `ignoreResizes`
+    if (
+      this.initialScrollWidth_ === null ||
+      this.initialScrollHeight_ === null
+    ) {
+      this.initialScrollWidth_ = layoutRect.width;
```

Comment:
One thing i want to double check is when is the best time to measure the `initialScrollHeight`. 
I have two concerns
1. scrollManager initiated too late that the layoutRect.height has already increased
2. scrollManager initiated too early, before `DOMContentLoaded` that the document height is still growing.

Reply:
(1): For the purposes of `<amp-next-page>` this is alright because the "initial" height is essentially the height before `<amp-next-page>` starts adding documents (which doesn't happen before the user scrolls). But I agree that for the general use case this is not ideal.
(2): I don't believe that's possible because scroll-manager is created after the `amp-analytics` element is built, right?